script.py

```diff
@@ -0,0 +1,186 @@
+        .select(f.col('id'))
+        .withColumnRenamed('id', 'MOLECULE_CHEMBL_ID')
+    )
+
+    # ----- GET PDB ID -----
+    # Unichem molecules DataFrame
+    unichem_df = spark.read.csv(args.unichem, sep=r'\t', header=True)
+
+    # Join Unichem and Molecule DataFrame
+    unichem_molecule_df = (molecule_df
+                           .join(unichem_df, unichem_df["From src:'1'"] == molecule_df['MOLECULE_CHEMBL_ID'])
+                           .withColumnRenamed("To src:'3'", 'MOLECULE_PDB_ID')
+                           .drop("From src:'1'")
+                           )
+
+    # TIMER 2 START
+    start_2 = timeit.default_timer()
+
+    # ----- GET STRUCTURE ID -----
+    # Apply get_structure function on the Unichem-Molecule DataFrame
+
+    unichem_molecule_struct_pd_df = (unichem_molecule_df
+                                     .toPandas()
+                                     .assign(STRUCTURE_ID=lambda x: x['MOLECULE_PDB_ID'].apply(get_structure))
+                                     )
+
+    unichem_molecule_struct_pd_df.to_csv('structure_of_molecules.csv', index=False, header=True)
+
+    # Convert Spark DataFrame into Pandas DataFrame
+    unichem_molecule_struct_spark_df = spark.createDataFrame(unichem_molecule_struct_pd_df)
+
+    # Count number of structure per molecule
+    unichem_molecule_struct_spark_df = (unichem_molecule_struct_spark_df
+                                        .withColumn("NB_OF_STRUCT_PER_MOL", f.size(f.col("STRUCTURE_ID")))
+                                        .orderBy(f.desc("NB_OF_STRUCT_PER_MOL"))
+                                        )
+
+    unichem_molecule_struct_spark_df = (unichem_molecule_struct_spark_df
+                                        .filter(f.col('MOLECULE_CHEMBL_ID') != 'CHEMBL692')
+                                        .filter(f.col('MOLECULE_CHEMBL_ID') != 'CHEMBL1236970')
+                                        .filter(f.col('MOLECULE_CHEMBL_ID') != 'CHEMBL457299')
+                                        .filter(f.col('MOLECULE_CHEMBL_ID') != 'CHEMBL113178')
+                                        .select('MOLECULE_CHEMBL_ID', 'MOLECULE_PDB_ID', 'STRUCTURE_ID'))
+
+    # Save into a CSV
+    unichem_molecule_struct_spark_df.toPandas().to_csv("molecule_structure_target.csv", index=False, header=True)
+
+    # TIMER 2 STOP
+    stop_2 = timeit.default_timer()
+
+    # ----- GET TARGET ID -----
+    pdb_target_df = create_pdb_target_gene_df(args.ensembl, unichem_molecule_struct_spark_df)
+
+    # ----- STATISTICS -----
+    total_nb_unichem = unichem_df.count()
+    total_nb_molecule = molecule_df.count()
+    nb_mol_in_unichem = unichem_molecule_df.count()
+
+    total_nb_struct = unichem_molecule_struct_spark_df.withColumn('STRUCTURE_ID', f.explode('STRUCTURE_ID')).select(
+        'STRUCTURE_ID').distinct().count()
+
+    nb_target_pd = pdb_target_df.count()
+    nb_human_target = pdb_target_df.filter(pdb_target_df.GENE_ID.startswith('ENSG')).count()
```

Comment:
When referring to columns it is nicer to use the `col` function eg:
```suggestion
    nb_human_target = pdb_target_df.filter(col('GENE_ID').startswith('ENSG')).count()
```

Reply:
#### Ok thank you! Do you know why it's better?

Reply:
It's better, not necessarily for better performance, but your expression will become you won't need to repeat the name of the dataframe. (Think about changing the name of the dataframe)